packages/protocol/test/stability/grandamento.ts

```diff
@@ -396,23 +456,556 @@ contract('GrandaMento', (accounts: string[]) => {
+            assertEqualBN(grandaMentoBalanceBefore.minus(grandaMentoBalanceAfter), sellAmountValue)
+            assertEqualBN(totalSupplyBefore.minus(totalSupplyAfter), sellAmountValue)
+          })
+
+          it('burns the entire stable token balance if the sell amount value is higher than the balance', async () => {
+            const newGrandaMentoBalance = unit.times(400)
+            // Transfer some stable token out of granda mento to simulate a situation
+            // where granda mento has a balance < the sell amount
+            await stableToken.transferFrom(
+              grandaMento.address,
+              owner,
+              stableTokenSellAmount.minus(newGrandaMentoBalance)
+            )
+
+            const totalSupplyBefore = await stableToken.totalSupply()
+            await grandaMento.executeExchangeProposal(1)
+            const totalSupplyAfter = await stableToken.totalSupply()
+
+            assertEqualBN(await stableToken.balanceOf(grandaMento.address), 0)
+            assertEqualBN(totalSupplyBefore.minus(totalSupplyAfter), newGrandaMentoBalance)
+          })
+
+          it('transfers the buyAmount of CELO out of the Reserve to the exchanger', async () => {
+            const exchangeProposal = parseExchangeProposal(await grandaMento.exchangeProposals(1))
+            const reserveBalanceBefore = await goldToken.balanceOf(reserve.address)
+            const exchangerBalanceBefore = await goldToken.balanceOf(alice)
+            await grandaMento.executeExchangeProposal(1)
+            const reserveBalanceAfter = await goldToken.balanceOf(reserve.address)
+            const exchangerBalanceAfter = await goldToken.balanceOf(alice)
+
+            assertEqualBN(
+              reserveBalanceBefore.minus(reserveBalanceAfter),
+              exchangeProposal.buyAmount
+            )
+            assertEqualBN(
+              exchangerBalanceAfter.minus(exchangerBalanceBefore),
+              exchangeProposal.buyAmount
+            )
+          })
+        })
+
+        describe('when selling CELO', () => {
+          before(() => {
+            sellCelo = true
+          })
+
+          it('transfers the CELO to the Reserve', async () => {
+            const grandaMentoBalanceBefore = await goldToken.balanceOf(grandaMento.address)
+            const reserveBalanceBefore = await goldToken.balanceOf(reserve.address)
+            await grandaMento.executeExchangeProposal(1)
+            const grandaMentoBalanceAfter = await goldToken.balanceOf(grandaMento.address)
+            const reserveBalanceAfter = await goldToken.balanceOf(reserve.address)
+
+            assertEqualBN(reserveBalanceAfter.minus(reserveBalanceBefore), celoSellAmount)
+            assertEqualBN(grandaMentoBalanceBefore.minus(grandaMentoBalanceAfter), celoSellAmount)
+          })
+
+          it('transfers the entire CELO balance to the Reserve if the sell amount is higher than the balance', async () => {
```

Comment:
In what scenario this would be hit?

Reply:
I don't see a situation tbh, that's why I'm on the fence about https://github.com/celo-org/celo-monorepo/pull/8082/files#diff-53b0c8ae012f6e9215f6c0216a5c6bbf74d4abce2fd1c1cd636e502d76be7608R353. It's a pretty common practice to have that sort of safety just in case to ensure that the call will never revert from an unforeseen error, though. I'm curious to hear Yorke/CAP's opinion